octodns/processor/clamp.py

```diff
@@ -0,0 +1,71 @@
+from logging import getLogger
+
+from .base import BaseProcessor, ProcessorException
+
+class TTLArgumentException(ProcessorException):
+    pass
+
+
+class TtlClampProcessor(BaseProcessor):
+    """
+    Processor that clamps TTL values to a specified range.
+
+    Configuration:
+        min_ttl: Minimum TTL value (default: 300 seconds / 5 minutes)
+        max_ttl: Maximum TTL value (default: 86400 seconds / 24 hours)
+
+    Example config.yaml:
+        processors:
+          clamp:
+            class: octodns.processor.clamp.TtlClampProcessor
+            min_ttl: 300
+            max_ttl: 3600
+
+        zones:
+          example.com.:
+            sources:
+              - config
+            processors:
+              - clamp
+            targets:
+              - route53
+    """
+
+    def __init__(self, id, min_ttl=300, max_ttl=86400):
+        super().__init__(id)
+        self.log = getLogger(
+            f'{self.__class__.__module__}.{self.__class__.__name__}'
+        )
+        if not min_ttl <= max_ttl:
+            raise TTLArgumentException(f'Min TTL {min_ttl} is not lower than max TTL {max_ttl}')
+        self.min_ttl = min_ttl
+        self.max_ttl = max_ttl
+        self.log.info(
+            f'TtlClampProcessor initialized: min={min_ttl}s, max={max_ttl}s'
```

Comment:
logging prefers placeholders rather than f-strings:

```suggestion
            f'__init__: min=%ss, max=%ss', min_ttl, max_ttx)
```